src/main/java/kr/sesac/aoao/server/user/domain/User.java

```diff
@@ -1,33 +1,48 @@
 package kr.sesac.aoao.server.user.domain;
 
+import org.springframework.security.crypto.password.PasswordEncoder;
+
 import kr.sesac.aoao.server.user.controller.dto.request.SignUpRequest;
 import kr.sesac.aoao.server.user.repository.Role;
 import lombok.Builder;
 import lombok.Getter;
 
 @Getter
 public class User {
+	private final Long userId;
 	private final String email;
 	private final String nickname;
 	private final String password;
 	private final String profile;
 	private final Role role;
 
 	@Builder
-	public User(String email, String nickname, String password, String profile, Role role) {
+	public User(Long userId, String email, String nickname, String password, String profile, Role role) {
+		this.userId = userId;
 		this.email = email;
 		this.nickname = nickname;
 		this.password = password;
 		this.profile = profile;
 		this.role = role;
 	}
 
-	public static User from(SignUpRequest signUpRequest){
+	public static User from(SignUpRequest signUpRequest, PasswordEncoder passwordEncoder) {
```

Comment:
```suggestion
	public static User from(SignUpRequest signUpRequest, String password) {
```
도메인 객체에 PasswordEncoder 를 주입해서 하는 방식보다는 암호화를 한 패스워드를 입력받는 건 어떨까요? 객체지향 적이지 않아 고민된다면 Password 를 암호화하는 유틸성 객체를 따로 생성 후 도메인끼리 메시지를 보내 통신하는 방법도 좋을 것 같습니다!